packages/datadog-plugin-bluebird/src/index.js

```diff
@@ -2,25 +2,56 @@
 
 const tx = require('../../dd-trace/src/plugins/util/promise')
 
-function createGetNewLibraryCopyWrap (tracer, config, shim) {
+const DD_LIB_COPIES = '_datadog_library_copies'
+
+function createGetNewLibraryCopyWrap (tracer, config, originalLib, shim) {
   return function wrapGetNewLibraryCopy (getNewLibraryCopy) {
     return function getNewLibraryCopyWithTrace () {
-      const result = getNewLibraryCopy.apply(this, arguments)
-      shim.wrap(result.prototype, '_then', tx.createWrapThen(tracer, config))
-      return result
+      const libraryCopy = getNewLibraryCopy.apply(this, arguments)
+      shim.wrap(libraryCopy.prototype, '_then', tx.createWrapThen(tracer, config))
+      addToLibraryCopies(originalLib.prototype, libraryCopy)
+      return libraryCopy
     }
   }
 }
 
+function addToLibraryCopies (originalLibPrototype, libraryCopy) {
+  let libraryCopies = originalLibPrototype[DD_LIB_COPIES]
+
+  if (!libraryCopies) {
+    libraryCopies = new Set()
+
+    Object.defineProperty(originalLibPrototype, DD_LIB_COPIES, {
+      enumerable: false,
+      writable: true
```

Comment:
You can set the value here rather than on line 29 with `value: libraryCopies`.

You should also add `configurable: true`.

`enumerable: false` is the default, so you can omit it if you want. (Your choice.)

Reply:
makes sense, updated with `configurable` and `value` and using the `enumerable` default